### DISCUSSION TITLE
XFRM configuration

### DISCUSSION THREAD

**Discussion by FB9pq** [strongswan/strongswan#1833] [Q&A] [answered]
Hello!
What tools are used to configure XFRM devices?
Documentation mentions only a temporary config
https://docs.strongswan.org/docs/5.9/features/routeBasedVpn.html#_example_create_xfrm_interface_ipsec0
It needs to be applied after every reboot.
I only saw brief mentions about etc/systemd/network/27-xfrm.netdev and /etc/systemd/network/27-xfrm.netdev configuration files.
Some spoke about a networkmanager plugin. But there is only a tool to add VTI.
Is there a confuguration example for an XFRM device that is integrated with network services and does not need to be entered manually on every boot?

**Comment by ptchuba** [comment, marked as answer]
If you are trying to do something like this:
```
sudo ip link add xfrm1 type xfrm dev lo if_id 1
sudo ip link set xfrm1 up
sudo ip addr add 10.10.0.1/24 dev xfrm1
sudo ip route add 10.0.0.0/8 dev  xfrm1
```

You can do this:
```
sudo tee /etc/systemd/network/xfrm1.netdev > /dev/null <<\EOF
[NetDev]
Name=xfrm1
Kind=xfrm

[Xfrm]
InterfaceId=1
EOF

sudo tee /etc/systemd/network/ipsec-assign.network > /dev/null <<\EOF
[Match]
Name=lo

[Network]
Xfrm=xfrm1
EOF

sudo tee /etc/systemd/network/xfrm1-route.network > /dev/null <<\EOF
[Match]
Name=xfrm1

[Network]
Address=10.10.0.1/24

[Route]
Destination=10.0.0.0/8
Scope=link
EOF

sudo systemctl restart systemd-networkd

```

**Reply by FB9pq** [reply]
Yes, this is a very useful option in case we are using systemd to manage network interfaces.
It should to be mentioned in documentation.
Also for Network Manager there is a directory /etc/NetworkManager/dispatcher.d/
If we place an executable script with only root permissions in there:

#!/bin/bash
ip link add ipsec0 type xfrm dev eth0 if_id 42
ip link set ipsec0 up
ip route add 10.0.0.0/24 dev ipsec0

Then we will have an ipsec interface configured on every boot

Thank you very much!
